core/modules/system/system.install

```diff
@@ -2505,7 +2505,34 @@ function system_update_1026() {
   $config->set('file_temporary_path', update_variable_get('file_temporary_path', ''));
   $config->set('file_private_path', update_variable_get('file_private_path', ''));
   $config->set('image_jpeg_quality', update_variable_get('image_jpeg_quality', 75));
-  $config->set('install_profile', update_variable_get('install_profile', 'standard'));
+
+  // Get the Drupal install profile and switch it to "standard" if not usable
+  // with Backdrop.
+  $current_profile = update_variable_get('install_profile', 'standard');
+  if ($current_profile != 'standard') {
+    $core_profiles = file_scan_directory('./core/profiles', '/\.profile$/', array('key' => 'name'));
+    $root_profiles = file_scan_directory('./profiles', '/\.profile$/', array('key' => 'name'));
+    $available = array_merge($core_profiles, $root_profiles);
```

Comment:
Is there a reason we can't simplify this by using [`backdrop_system_listing()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/backdrop_system_listing/1) instead?

Seems to provide the same output in my testing:

![image](https://user-images.githubusercontent.com/2385329/221099856-d79b64b0-5781-4b9a-9c36-cf1e4b882c54.png)

```suggestion
    $available = backdrop_system_listing('/\.profile$/', 'profiles');
```